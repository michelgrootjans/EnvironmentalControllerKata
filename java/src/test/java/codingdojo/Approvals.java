package codingdojo;

import org.approvaltests.reporters.ClipboardReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.jupiter.api.Test;

import static org.approvaltests.combinations.CombinationApprovals.verifyAllCombinations;

public class Approvals {
    @Test
    @UseReporter(ClipboardReporter.class)
    void approvals() {
        verifyAllCombinations(
                this::applyController,
                new Integer[]{5, 6, 15, 16, 24, 25, 29, 30}
                );
    }

    private Hardware applyController(int temperature) {
        HardwareStub hw = new HardwareStub();
        EnvironmentController controller = new EnvironmentController(hw);
        hw.setTemp(temperature);
        controller.tic();
        return hw;
    }

}
