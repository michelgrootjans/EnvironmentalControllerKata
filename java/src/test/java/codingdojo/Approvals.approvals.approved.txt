[5] => {blower: 'on', heater: 'on', loTempAlarm: 'on', cooler: 'off', hiTempAlarm: 'off'} 
[6] => {blower: 'on', heater: 'on', loTempAlarm: 'off', cooler: 'off', hiTempAlarm: 'off'} 
[15] => {blower: 'on', heater: 'on', loTempAlarm: 'off', cooler: 'off', hiTempAlarm: 'off'} 
[16] => {blower: 'off', heater: 'off', loTempAlarm: 'off', cooler: 'off', hiTempAlarm: 'off'} 
[24] => {blower: 'off', heater: 'off', loTempAlarm: 'off', cooler: 'off', hiTempAlarm: 'off'} 
[25] => {blower: 'on', heater: 'off', loTempAlarm: 'off', cooler: 'on', hiTempAlarm: 'off'} 
[29] => {blower: 'on', heater: 'off', loTempAlarm: 'off', cooler: 'on', hiTempAlarm: 'off'} 
[30] => {blower: 'on', heater: 'off', loTempAlarm: 'off', cooler: 'on', hiTempAlarm: 'on'} 
